cmd/main.go

```diff
@@ -86,6 +86,15 @@ func main() {
 					return
 				}
 			}
+		} else {
+			if d.Mode.Artifact == "" {
```

Comment:
The Mode.Artifact probably should be renamed: Mode.Binary as it is the path to the binary within the task bucket.  Not to be confused with the Application.Binary which contains maven coordinates to the binary.

Anyway, when there isn't an uploaded binary to analyze, we need to run maven to pull down the binary using the coordinates.